### PR TITLE
ENH: Show colors in terminology selector

### PR DESCRIPTION
**Pull request by cpinter**
The recommended colors for are shown as a color square for each type in the terminology navigator widget.

image

### CODE REVIEW

File: Modules/Loadable/Terminologies/Widgets/qSlicerTerminologyNavigatorWidget.cxx [hunk 1]
```diff
@@ -1170,49 +1170,62 @@ void qSlicerTerminologyNavigatorWidget::populateTypeTable()
...
-    }
-    else
-    {
-      d->tableWidget_Type->setRowCount(types.size());
-    }
+  // Show none item only if search term is empty (if user is searching then they want an actual type)
+  int noneTypeExists = 0;
+  QTableWidgetItem* noneItem = nullptr;
+  if (searchTerm.empty())
+  {
```
Comment (lassoan) [inline Modules/Loadable/Terminologies/Widgets/qSlicerTerminologyNavigatorWidget.cxx:1179]: indentation

**Review by lassoan — APPROVED** [review]
This looks great, thank you!

**Comment by cpinter** [comment]
I fixed the indentation and added a check not to show color if the type did not have a color to begin with, like this
image

**Comment by lassoan** [comment]
When does it happen that the type does not have a color? When it has a modifier? In that case, it would be better to show the first available color (usually they are the same or very similar for all modifiers).

**Comment by cpinter** [comment]
Yes it happens when they have modifiers. I think I can get the first modifier. Not sure how much it will slow things down, probably not noticeably.

**Comment by cpinter** [comment]
Done, the first modifier's color is used if found.

**Review by lassoan — CHANGES_REQUESTED** [review]
This will be good, there is just a build error on linux according to CircleCI.

**Comment by cpinter** [comment]
I fixed the error.

**Review by lassoan — APPROVED** [review]

**Comment by lassoan** [comment]
Thank you!

**Pull request merged**